src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -658,4 +662,23 @@ private long implTransferTo(OutputStream out) throws IOException {
         }
     }
 
+    /**
+     * Returns true if this class satisfies two conditions:
+     * <ul>
+     * <li>the reference to {@code byte[]} is not kept within the class</li>
+     * <li>the argument of {@link OutputStream#write(byte[])}} and {@link OutputStream#write(byte[], int, int)}} is not modified within the methods</li>
+     * <li>the {@code byte[]} is not read outside of the given bounds</li>
+     * </ul>
+     *
+     * @return true if this class is trusted
+     * @see java.io.ByteArrayInputStream#transferTo(OutputStream)
+     * @see java.io.BufferedInputStream#implTransferTo(OutputStream)
+     */
+    private static boolean isTrusted(OutputStream os) {
```

Comment:
Let's follow Alan's proposal and simply inline this. The solution is in sync with BAIS then, and we can merge it quickly.

Reply:
> Let's follow Alan's proposal and simply inline this. 

I didn't suggest inlining this. I just asked not to add a new package com.sun.io with a single method class. Also as the concerns with BAIS and BIS are not the same (they do overlap but they are not the same), I think it's best to keep the changes as local as possible. It can be made more widely useful in the future but great care must be taken.

Reply:
> > Let's follow Alan's proposal and simply inline this.
> 
> I didn't suggest inlining this. I just asked not to add a new package com.sun.io with a single method class. Also as the concerns with BAIS and BIS are not the same (they do overlap but they are not the same), I think it's best to keep the changes as local as possible. It can be made more widely useful in the future but great care must be taken.

Well, effectively inlining is the *ultimate* locality. ;-)